lib/ui/screens/types/modal_contents.dart

```diff
@@ -16,18 +25,138 @@ class ModalContents extends StatelessWidget {
   final int index;
   final double width;
   final ScrollController scroller;
+
+  @override
+  State<ModalContents> createState() => _ModalContentsState();
+}
+
+class _ModalContentsState extends State<ModalContents> {
+  final List<bool> _isOpen = [false, false, false];
+
+  void _onPokemonPress(int index, Pokemon pokemon) {
+    context.read(currentPokemonStateProvider).setPokemon(index, pokemon);
+
+    print(pokemon.name);
+
+    AppNavigator.push(Routes.pokemonInfo, pokemon);
+  }
+
+  ExpansionPanel _buildTypePokemonPanel(List<Pokemon> pokemons) {
+    final pokemonsFiltered = pokemons
+        .where((pokemon) => pokemon.types.contains(types[widget.index].type))
```

Comment:
```dart
types[widget.index].type)
```
Should we handle the case where `types[widget.index]` doesn't exist?

Reply:
Also, you can extract the types[widget.index] to something like this since I see you're using it in multiple places

```dart
<Type> get type => types[widget.index]
```

Reply:
What do you want to do if there is no type ? Do I let the expansion panel visible ?

Reply:
I think types[widget.index] can't doesn't exist, because the index is set by selecting a type in the types list, so it's impossible to have a null type. But if you want to add this to respect the NULL SAFETY it's not a problem I can just hide the expansion panel 😄

Reply:
@Matttx Hmm I didn't notice that the `types[widget.index]` was being used across this page at the beginning. Let just leave it for now and I will rewrite this page in the future